### PR TITLE
Remove deprecated knative/build support

### PR DESCRIPTION
**Pull request by eddycharly**
This pull request is work in progress.

The knative/build is deprecated and can be removed as discussed here https://github.com/kubernetes/test-infra/issues/13948

This pull request also contains changes proposed in https://github.com/kubernetes/test-infra/issues/14983

There's still a lot of work to do, i create the pull request early in case someone wants to give some feedback.

TODOS
- Ensure knative/build is completely removed
- Fix/adapt tests
- Fix doc

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/ok-to-test

There is btw also https://github.com/kubernetes/test-infra/pull/14983 which removes just the manifests

File: prow/cmd/build/controller.go [hunk 1]
```diff
@@ -17,30 +17,23 @@ limitations under the License.
 package main
```
Comment (alvaroaleman) [inline prow/cmd/build/controller.go]: Just for my understanding, why do we keep anything of this controller?

Comment (eddycharly) [reply prow/cmd/build/controller.go]: i'm not sure if it is completely knative/build specific, isn't it the prow-pipeline itself ?

**Comment by eddycharly** [comment]
@alvaroaleman thanks, i will reintegrate the changes in #14983

File: prow/cmd/build/controller.go (hunk 1 above)
Comment (alvaroaleman) [reply prow/cmd/build/controller.go]: @fejta would be the authoritative source to confirm this but I believe so. The pipeline controller is in `prow/cmd/pipeline`

Comment (eddycharly) [reply prow/cmd/build/controller.go]: You're rignt, we can remove it.

**Comment by eddycharly** [comment]
Bazel suddenly stopped building, whatever i ask bazel now i have this error
```
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
INFO: Writing tracer profile to '/private/var/tmp/_bazel_charlesbreteche/42dc72a11735a84e3aad14550c4ca981/command.profile.gz'
ERROR: Failed to load Starlark extension '@rules_proto//proto:defs.bzl'.
Cycle in the workspace file detected. This indicates that a repository is used prior to being defined.
The following chain of repository dependencies lead to the missing definition.
 - @rules_proto
This could either mean you have to add the '@rules_proto' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing defini
tion earlier in your WORKSPACE file.
ERROR: cycles detected during target parsing
INFO: Elapsed time: 25.773s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
```

It worked perfectly a few hours before.
Tried to solve this by myself but i don't know bazel well and couldn't find a simple solution.

Strange it stopped working suddenly.
Any idea about how to fix this ?

**Comment by fejta** [comment]
What version of bazel are you using? From the error message it sounds like something wants https://github.com/bazelbuild/rules_proto to exist, which now doesn't, for unknown reasons.

You could try adding this to WORKSPACE, or you could try ensuring that you're using an expected https://github.com/kubernetes/test-infra/blob/3ad2eb8c4f5a65f8842041cb46920d1b106b02d2/.bazelversion#L1 by using something like bazelisk: https://github.com/bazelbuild/bazelisk#requirements

**Comment by eddycharly** [comment]
Started from a fresh git clone, the previous has gone away.

Trying to fix `pull-test-infra-bazel` job, i ran `bazel run @io_k8s_repo_infra//hack:update-bazel` then `bazel run @io_k8s_repo_infra//hack:update-deps`.

The latter fails with
```
$ bazel run @io_k8s_repo_infra//hack:update-deps
INFO: Build option --platforms has changed, discarding analysis cache.
INFO: Analyzed target @io_k8s_repo_infra//hack:update-deps (63 packages loaded, 7445 targets configured).
INFO: Found 1 target...
Target @io_k8s_repo_infra//hack:update-deps up-to-date:
  bazel-bin/external/io_k8s_repo_infra/hack/update-deps
INFO: Elapsed time: 13.550s, Critical Path: 0.31s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/external/io_k8s_repo_infra/hack/update-deps external/go_sdk/bin/go external/io_k8s_repo_infra/hack/update-bazel external/com_github_bazelbuild_buildtools/buildifier/darwin_amd64_striINFO: Build completed successfully, 1 total action
Updating modules...
warning: ignoring symlink /Users/charlesbreteche/Documents/dev/test-infra/bazel-bin
warning: ignoring symlink /Users/charlesbreteche/Documents/dev/test-infra/bazel-genfiles
warning: ignoring symlink /Users/charlesbreteche/Documents/dev/test-infra/bazel-out
warning: ignoring symlink /Users/charlesbreteche/Documents/dev/test-infra/bazel-test-infra
warning: ignoring symlink /Users/charlesbreteche/Documents/dev/test-infra/bazel-testlogs
go: finding golang.org/x/mobile latest
go: finding github.com/elazarl/goproxy latest
go: finding gonum.org/v1/netlib latest
go: finding github.com/kr/logfmt latest
go: finding github.com/xiang90/probing latest
go: finding github.com/google/pprof latest
go: finding github.com/armon/consul-api latest
go: finding k8s.io/component-base latest
go: finding github.com/asaskevich/govalidator latest
go: finding github.com/mxk/go-flowrate latest
go: finding github.com/alecthomas/units latest
go: finding k8s.io/apiserver latest
go: finding github.com/globalsign/mgo latest
go: finding github.com/BurntSushi/xgb latest
go: finding github.com/coreos/pkg latest
go: finding github.com/docker/spdystream latest
go: finding github.com/remyoudompheng/bigfft latest
go: finding github.com/mitchellh/ioprogress latest
go: finding github.com/coreos/go-systemd latest
go: finding golang.org/x/image latest
go: finding github.com/docopt/docopt-go latest
go: finding gopkg.in/yaml.v1 latest
go: finding github.com/tmc/grpc-websocket-proxy latest
go: finding github.com/munnerz/goautoneg latest
go: finding github.com/alecthomas/template latest
go: finding github.com/pquerna/cachecontrol latest
go: finding golang.org/x/build latest
go: finding github.com/mwitkow/go-conntrack latest
go: finding sigs.k8s.io/structured-merge-diff latest
Updating bazel rules...
FAILED
```

**Comment by eddycharly** [comment]
I fixed the update-deps error, it was related to macos not having realpath.
`brew install coreutils` fixed the issue.

Now i'm blocked with this new problem https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/15022/pull-test-infra-bazel/1190264326607343616

Looks like a dependency is not correctly declared...

**Comment by eddycharly** [comment]
@fejta @alvaroaleman Completely lost on this error, can someone help please ?

**Comment by alvaroaleman** [comment]
@eddycharly dependencies are a mess and I can not explain to you why, but the following fixes the issue:

```
# Reset dependency declarations to what is in origin/master
git fetch origin
git checkout origin/master go.mod go.sum repos.bz
./hack/update-deps.sh
```

There seems to be an issue with the updated  `cloud.google.com/go v0.45.1` and bazel not being able to cope with that. I know very little about bazel so I can not give an explanation either :(

**Comment by eddycharly** [comment]
@alvaroaleman 👍it just worked, thanks !

I was reading https://github.com/bazelbuild/bazel-gazelle/issues/656

I am very new to bazel, gazelle, and all the ecosystem... i can find my way when it works but when it fails it's somewhat difficult to understand why :(

**Comment by alvaroaleman** [comment]
@eddycharly gret to haer, I also know very little about bazel, this was just guessing based on the fact that the dependency that caused issue was bumped :)

Could you also squash your commits together so its only one as described [https://github.com/wprig/wprig/wiki/How-to-squash-commits](here)?

File: repos.bzl [hunk 2]
```diff
@@ -834,8 +834,8 @@ def go_repositories():
         build_file_generation = "on",
         build_file_proto_mode = "disable",
         importpath = "github.com/knative/build",
-        sum = "h1:3KvuoUpgXYkK3WKaSBVN/0FSwRaukf212t2bQkMSWjY=",
-        version = "v0.3.1-0.20190330033454-38ace00371c7",
+        sum = "h1:o/VYWA3HKyZlNqdU2hDE5LHpanBe8gazgPKL97XJ6bo=",
```
Comment (alvaroaleman) [inline repos.bzl]: @fejta Why is this getting bumped instead of removed?

Comment (eddycharly) [reply repos.bzl]: It can be removed now, it came back when i fetched files from master to fix the dependencies issue.

Comment (fejta) [reply repos.bzl]: Because that is what is in go.sum

Comment (eddycharly) [reply repos.bzl]: @fejta i removed knative/build from go.sum and repos.bzl but now the job fails (https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/15022/pull-test-infra-bazel/1190347627145728003).
Should i re add them ?

**Comment by eddycharly** [comment]
/assign @bentheelder
The pull request should be ready for review now.
Thanks @fetja & @alvaroaleman for your help.

File: repos.bzl (hunk 2 above)
Comment (fejta) [reply repos.bzl]: Yes, I would just run `hack/update-deps.sh` and let it do its thing.


Something (most likely parts of pipeline) are depending on knative and therefore knative/build. Something like `go mod graph | grep /build` will help you dig into this (looks like aws-k8s-tester depends on it, for example), but I wouldn't spend a lot of time on this

File: repos.bzl [hunk 3]
```diff
@@ -2187,3 +2179,11 @@ def go_repositories():
         sum = "h1:u2CU3YKy9I2pmu9pX0eq50wCgjfGIt539SqR7FbHiho=",
         version = "v1.0.4",
     )
+    go_repository(
+        name = "com_github_knative_build",
```
Comment (fejta) [inline repos.bzl]: Idelaly this will stay above the `com_github_knative_pkg repo` where it has been.

Comment (eddycharly) [reply repos.bzl]: done

**Comment by fejta** [comment]
@cjwagner we'll need to `for i in deployment rbac; do bazel run //prow/cluster:build_${i}.delete || break; done` prior to this commit (after it merges to delete the jobs).

Let me know if you want me to do this

**Comment by BenTheElder** [comment]
whoo!

**Comment by cjwagner** [comment]
Looks like the prow components have been removed, but the actual build controller in the `knative-build` namespace still needs to be removed. I'm going to delete that namespace.

**Pull request merged**